component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/handlers/MTLSTokenBindingRefreshGrantHandler.java

```diff
@@ -0,0 +1,108 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls.handlers;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2AccessTokenRespDTO;
+import org.wso2.carbon.identity.oauth2.token.OAuthTokenReqMessageContext;
+import org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls.utils.CommonConstants;
+import org.wso2.carbon.identity.oauth2.token.handlers.grant.RefreshGrantHandler;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * If MTLS token binding is used (MTLSTokenBindingAuthorizationCodeGrantHandler), the certificate of the client is
+ * bounded to the access token using a hidden scope. This class is used to remove the hidden scope from the token
+ * response.
+ *
+ * @see <href="https://tools.ietf.org/html/draft-ietf-oauth-mtls-17">IETF OAuth MTLS</>
+ */
+public class MTLSTokenBindingRefreshGrantHandler extends RefreshGrantHandler {
+
+    private static final Log log = LogFactory.getLog(MTLSTokenBindingRefreshGrantHandler.class);
+
+    @Override
+    public OAuth2AccessTokenRespDTO issue(OAuthTokenReqMessageContext tokReqMsgCtx)
+            throws IdentityOAuth2Exception {
+
+        OAuth2AccessTokenRespDTO oAuth2AccessTokenRespDTO = super.issue(tokReqMsgCtx);
+        tokReqMsgCtx.setScope(getReducedResponseScopes(tokReqMsgCtx.getScope()));
+        return oAuth2AccessTokenRespDTO;
+    }
+
+    @Override
+    public boolean validateScope(OAuthTokenReqMessageContext tokReqMsgCtx)
+            throws IdentityOAuth2Exception {
+
+        String[] grantedScopes = tokReqMsgCtx.getScope();
+        if (!super.validateScope(tokReqMsgCtx)) {
+            return false;
+        }
+
+        String[] requestedScopes = tokReqMsgCtx.getOauth2AccessTokenReqDTO().getScope();
+        String[] modifiedScopes;
+        if (ArrayUtils.isNotEmpty(requestedScopes)) {
+            if (ArrayUtils.isEmpty(grantedScopes)) {
+                return false;
+            }
+
+            // Add cert hash scope from previously granted scopes.
+            ArrayList<String> requestedScopeList = new ArrayList<>(Arrays.asList(requestedScopes));
+            for (String scope : grantedScopes) {
+                if (isAllowedScope(scope)) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Adding custom scope " + scope + " to the requested scopes");
```

Comment:
Add a full stop to the end of the sentence.

Reply:
Fixed https://github.com/wso2-extensions/identity-oauth-addons/pull/56/commits/e4075860c2749e90d901b133399a9dacfa441376